packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,451 @@
+      },
+    ],
+    [
+      "Product Collection Title",
+      (line: Product): string => line.collection.title,
+    ],
+    [
+      "Product Collection Handle",
+      (line: Product): string => line.collection.handle,
+    ],
+    ["Product Type", (line: Product): string => line.type.value],
+    [
+      "Product Tags",
+      (line: Product): string =>
+        (line.tags.map((t) => t.value) ?? []).join(","),
+    ],
+    [
+      "Product Discountable",
+      (line: Product): string => line.discountable?.toString(),
+    ],
+    ["Product External ID", (line: Product): string => line.external_id],
+    ["Product Profile Name", (line: Product): string => line?.profile?.name],
+    ["Product Profile Type", (line: Product): string => line?.profile?.type],
+    /*
+     *
+     * The other columns are dynamically computed via the `buildHeader` method.
+     * This will ensure that all the expected columns based on the
+     * real data and therefore the highest number of value for the different
+     * relation that we are expecting to appear
+     *
+     */
+  ])
+
+  private readonly NEWLINE = "\r\n"
+  private readonly DELIMITED_ = ";"
+  private readonly BATCH_SIZE = 50
+
+  constructor({
+    manager,
+    batchJobService,
+    productService,
+    fileService,
+    productRepository,
+  }: InjectedDependencies) {
+    super({
+      manager,
+      batchJobService,
+      productService,
+      fileService,
+      productRepository,
+    })
+
+    this.manager_ = manager
+    this.batchJobService_ = batchJobService
+    this.productService_ = productService
+    this.fileService_ = fileService
+    this.productRepository_ = productRepository
+  }
+
+  async buildTemplate(): Promise<string> {
+    return this.buildHeader()
+  }
+
+  async completeJob(batchJobId: string): Promise<BatchJob> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      const batchJob = await this.batchJobService_
+        .withTransaction(transactionManager)
+        .retrieve(batchJobId)
+
+      if (batchJob.status !== BatchJobStatus.COMPLETED) {
+        return await this.batchJobService_.complete(batchJob)
+      } else {
+        return batchJob
+      }
+    })
+  }
+
+  async prepareBatchJobForProcessing(
+    batchJobId: string,
+    req: Express.Request
+  ): Promise<BatchJob> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      // TODO: waiting for #1593
+      /*
+        const batchJob = this.batchJobService_
+          .withTransaction(transactionManager)
+          .retrieve(batchJobId)
+
+        const { listConfig, filterableFields } = transformQuery({ defaultRelations: relations_ ,isList: true})(req, {} as any, (err) => throw err)
+        batchJob.context = {
+          ...(batchJob.context ?? {}),
+          listConfig,
+          filterableFields
+        }
+
+        const batchJobRepo = transactionManager.getCustomRepository(this.batchJobRepository_)
+        batchJobRepo.save(batchJob)
```

Comment:
You should be able to move this into `validateContext` since the context returned(the validated context is written into the batch job when it's created 😄

Reply:
Yes true 👌

Reply:
then should we rename that method `prepareContext` which includes the validation?